src/capi/abstract.cpp

```diff
@@ -2071,8 +2071,7 @@ extern "C" PyObject* PyNumber_Long(PyObject* o) noexcept {
         return res;
     }
     if (PyLong_Check(o)) { /* A long subclass without nb_long */
-        BoxedInt* lo = (BoxedInt*)o;
-        return PyLong_FromLong(lo->n);
+        return o;
```

Comment:
CPython does `return _PyLong_Copy(o);`.  I can't find any good way to trigger this case (I think it only happens via an extension class?) but we might as well do it that way?